packages/orbit-components/src/Popover/index.tsx

```diff
@@ -113,8 +128,10 @@ const Popover = ({
         clearShownTimeout();
         setRenderWithTimeout(false);
       }
+      document.addEventListener("keydown", handleEsc);
```

Comment:
Why are you adding event listener also here since you already added it in `handleClick`? And you are not removing it on the outside click.

And why you just didn't add the event listener here in the effect, e.g. when Popover is shown, with cleanup function but you are doing it inside the `handleClick`? What is the advantage of doing it like this?

Reply:
I added `addEventListener` to `useEffect` as well, bcs we have to attach this listener in case when Popover is opened by default (prop `opened=true`). If I checked it correctly, when clicked/enter the Popover, only `handleClick` is called and I need to attach the eventListener there.